Simplified/Catalog/NYPLCatalogUngroupedFeed.m

```diff
@@ -195,14 +222,23 @@ - (void)prepareForBookIndex:(NSUInteger)bookIndex
        if(!ungroupedFeed) {
          NYPLLOG(@"Failed to fetch next page.");
          self.currentlyFetchingNextURL = NO;
+         if(handler) {
+           handler(self);
+         }
          return;
        }
        
        [self.books addObjectsFromArray:ungroupedFeed.books];
        self.nextURL = ungroupedFeed.nextURL;
        self.currentlyFetchingNextURL = NO;
        
-       [self prepareForBookIndex:self.greatestPreparationIndex];
+       if(handler) {
```

Comment:
I could be misunderstanding, but shouldn't the handler be called only when we are exiting/breaking the recursive loop?

Reply:
From my understanding, we are exiting here because we have found books that are supported and need to return the `NYPLCatalogUngroupedFeed` object to the view controller, in order to update the UI and set the delegates. The view controller will call `prepareForBookIndex:` if more books are needed.

Reply:
ah, so we are supposed to exit but then it's missing a `return` statement after the `handler` call maybe?